src/MainComponent.cpp

```diff
@@ -63,71 +74,94 @@ void MainComponent::RegisterLogLevel(EnergyPlus::Error log_level) {
 
 Element MainComponent::Render() {
   static int i = 0;
-  std::set<EnergyPlus::Error> allowed_level;
-  for (auto& [level, state] : level_checkbox) {
-    if (state) {
-      allowed_level.insert(level);
-    }
-  }
-
-  std::vector<ErrorMessage*> filtered_errorMsgs;
-  filtered_errorMsgs.reserve(m_errors.size());
-
-  bool prevAllowed = false;
-
-  for (auto& m_error : m_errors) {
-    if (allowed_level.contains(m_error.error)) {
-      filtered_errorMsgs.push_back(&m_error);
-      prevAllowed = true;
-    } else if (m_error.error == EnergyPlus::Error::Continue && prevAllowed) {
-      filtered_errorMsgs.push_back(&m_error);
-    }
-  }
 
   int current_line = (std::min(tab_selected_, 1) == 0 ? m_stdout_displayer : m_error_displayer)->selected();
 
-  auto header = hbox({
-    text(L"EnergyPlus-Cpp-Demo"),
-    filler(),
-    separator(),
+  if (tab_selected_ == 0) {
 
-    hcenter(toggle_->Render()),
-    separator(),
-    gauge(float(current_line) / float(std::max(1, (int)filtered_errorMsgs.size() - 1))) | color(Color::GrayDark),
-    separator(),
-    spinner(5, i++),
-  });
+    auto header = hbox({
+      text(L"EnergyPlus-Cpp-Demo"),
+      filler(),
+      separator(),
+      hcenter(toggle_->Render()),
+      separator(),
+      text(to_wstring(current_line)),
+      text(L"/"),
+      text(to_wstring(m_stdout_lines.size())),
+      separator(),
+      gauge(float(current_line) / float(std::max(1, (int)m_stdout_lines.size() - 1))) | color(Color::GrayDark),
+      separator(),
+      spinner(5, i++),
+      m_quitButton->Render(),
+    });
```

Comment:
Avoid doing work when it's not needed by rendering only needed components based on tabs